openpype/modules/shotgrid/plugins/publish/collect_shotgrid_entities.py

```diff
@@ -0,0 +1,103 @@
+import os
+
+import pyblish.api
+from pymongo import MongoClient
+
+from openpype.modules.shotgrid.lib.settings import (
+    get_shotgrid_project_settings,
+)
+
+
+class CollectShotgridEntities(pyblish.api.ContextPlugin):
+    """Collect shotgrid entities according to the current context"""
+
+    order = pyblish.api.CollectorOrder + 0.499
+    label = "Shotgrid entities"
+
+    def process(self, context):
+
+        avalon_project = context.data.get("projectEntity")
+        avalon_asset = context.data.get("assetEntity")
+        avalon_task_name = os.getenv("AVALON_TASK")
+
+        self.log.info(avalon_project)
+        self.log.info(avalon_asset)
+
+        sg_project = _get_shotgrid_project(avalon_project)
+        sg_task = _get_shotgrid_task(
+            avalon_project, avalon_asset, avalon_task_name
+        )
+        sg_entity = _get_shotgrid_entity(avalon_project, avalon_asset)
+
+        if sg_project:
+            context.data["shotgridProject"] = sg_project
+            self.log.info(
+                "Collected correspondig shotgrid project : {}".format(
+                    sg_project
+                )
+            )
+
+        if sg_task:
+            context.data["shotgridTask"] = sg_task
+            self.log.info(
+                "Collected correspondig shotgrid task : {}".format(sg_task)
+            )
+
+        if sg_entity:
+            context.data["shotgridEntity"] = sg_entity
+            self.log.info(
+                "Collected correspondig shotgrid entity : {}".format(sg_entity)
+            )
+
+    def _find_existing_version(self, code, context):
+
+        filters = [
+            ["project", "is", context.data.get("shotgridProject")],
+            ["sg_task", "is", context.data.get("shotgridTask")],
+            ["entity", "is", context.data.get("shotgridEntity")],
+            ["code", "is", code],
+        ]
+
+        sg = context.data.get("shotgridSession")
+        return sg.find_one("Version", filters, [])
+
+
+def _get_shotgrid_collection(project):
+    mongo_url = os.getenv("OPENPYPE_MONGO")
+    client = MongoClient(mongo_url)
```

Comment:
I would recommend to use `OpenPypeMongoConnection.get_mongo_client()` to avoid  creating multiple mongo connection, does not have to initialize new connection (`from openpype.lib.mongo import OpenPypeMongoConnection`). Also the goal is not to create Mongo connection on multiple places using different approaches. One reason may be issues connected to DCC missing ssl certificates.

Reply:
This might be a new feature, am I wrong? (@ClementHector)